pkg/plugin/interface.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"context"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+)
+
+// Handshake is a common handshake that is shared by plugin and host.
+// handshakeConfigs are used to just do a basic handshake between
+// a plugin and host. If the handshake fails, a user friendly error is shown.
+// This prevents users from executing bad plugins or executing a plugin
+// directory. It is a UX feature, not a security feature.
+var HandShake = plugin.HandshakeConfig{
+	// This isn't required when using VersionedPlugins
+	ProtocolVersion:  1,
+	MagicCookieKey:   "JIRA_PLUGIN",
+	MagicCookieValue: "hello",
+}
+
+// PluginMap is the map of plugins we can dispense.
+var PluginMap = map[string]plugin.Plugin{
+	"jira_plugin_grpc": &ValidatorGRPCPlugin{},
+}
```

Comment:
I think this map is static, and contains the full list of issue_plugin, while there is a separate []PluginConfig as demo-ed in the serve.go to select a subset from this map. Let me know whether that is the case. @sethvargo for more suggestions

Reply:
You generally define a plugin directory (config option) and then walk the directory and instantiate a plugin for each binary in the given directory. The Terraform and Vault examples are super complicated given some of the stuff they do, but the old Otto example might be useful: https://github.com/hashicorp/otto/blob/v0.2.0/command/plugin_manager.go

The API is slightly different now, but I think it will provide a good starting point. In short:

1. Accept a config `plugin_directory`
2. Walk over all executable binary files in that directory
3. For each file, create a `plugin.Plugin` and config
4. Put into a data structure

Reply:
File a separate PR https://github.com/abcxyz/jvs/pull/292 and follow this flow there.